compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/lower/inlineclasses/MemoizedInlineClassReplacements.kt

```diff
@@ -137,18 +137,19 @@ class MemoizedInlineClassReplacements(private val mangleReturnTypes: Boolean, pr
             require(function.dispatchReceiverParameter != null && function is IrSimpleFunction)
             val newValueParameters = ArrayList<IrValueParameter>()
             for ((index, parameter) in function.explicitParameters.withIndex()) {
```

Comment:
Maybe don't use `explicitParameters` at all now and handle dispatch/extension receivers outside of the loop (which can go only through `valueParameters`)?

Reply:
Good point, and the loop below doesn't really need to construct this list either.